compiler_gym/envs/llvm/__init__.py

```diff
@@ -12,7 +13,10 @@
     make_benchmark,
 )
 from compiler_gym.envs.llvm.llvm_env import LlvmEnv
-from compiler_gym.envs.llvm.specs import observation_spaces, reward_spaces
+
+if "compiler_gym.envs.llvm.is_making_specs" not in sys.modules:
+    from compiler_gym.envs.llvm.specs import observation_spaces, reward_spaces
```

Comment:
Woah, that's wild. I love it 😀 

I think it would be worth adding a TODO note that we can revisit later:

```suggestion
# TODO(github.com/facebookresearch/CompilerGym/issues/506): Tidy up.
if "compiler_gym.envs.llvm.is_making_specs" not in sys.modules:
    from compiler_gym.envs.llvm.specs import observation_spaces, reward_spaces
```

Reply:
Done.